src/DotNetLightning.Core/Serialize/Features.fs

```diff
@@ -230,6 +224,29 @@ type FeatureBit private (bitArray) =
     override this.ToString() =
         this.BitArray.PrintBits()
         
+    member this.SetFeatureBit(index: int, on: bool): unit =
+        let length = this.BitArray.Length
+        if length <= index then
+            this.BitArray.Length <- index + 1
+
+            //this.BitArray.RightShift(index - length + 1)
+
+            // NOTE: Calling RightShift gives me:
+            // "The field, constructor or member 'RightShift' is not defined."
+            // So I just re-implement it here
+            for i in (length - 1) .. -1 .. 0 do
+                this.BitArray.[i + index - length + 1] <- this.BitArray.[i]
+
+            // NOTE: this probably wouldn't be necessary if we were using
+            // RightShift, but the dotnet docs don't actualy specify that
+            // RightShift sets the leading bits to zero.
+            for i in 0 .. (index - length) do
+                this.BitArray.[i] <- false
+        this.BitArray.[this.BitArray.Length - index - 1] <- on
+
+    member this.GetFeatureBit(index: int): bool =
```

Comment:
Also, I suppose this should take `Feature` and `FeaturesSupport` too, and `IsFeatureOn` sounds better name. because it is more obvious that the return value is bool.

Reply:
And I think this method should be named `HasFeature` too,  because it is more obvious that the return value is bool.

Reply:
or `IsFeatureOn` you can choose.

Reply:
I've removed `GetFeature` and just use `HasFeature` instead.